Automattic/woocommerce-payments
# Confirm subscription owner before using its currency for switch purposes

#### Changes proposed in this Pull Request

Small change to verify the subscription being switched that is passed is owned by the current user. Without this check, it may be possible to change the currency during the switch process, resulting in an error on checkout.

Noting that unit tests do not account for logging due to the [Logger class is static](https://github.com/Automattic/woocommerce-payments/issues/611).

#### Testing instructions

Setup:
* Go to WooCommerce > Settings > Subscriptions and confirm that _Between Subscription Variations_ is enabled under Switching.
* Go to WooCommerce > Settings > Multi-Currency and enable at least one other currency.
* Create a new variable subscription with at least 2 variations if you do not have one already.
* Confirm logging is enabled under Payments > Settings > Advanced.

Testing:
* On your primary account, purchase one of the variations of the subscription product from setup using the default store currency.
* Once the confirmation screen shows, take note of the subscription ID.
* In an incognito/private browsing window, purchase the same variation under a guest account.
* On the order confirmation screen, click to view the subscription.
* Click the upgrade/downgrade button for the subscription.
* Once the page loads, update the url in the address bar replacing the ID after `switch-subscription` with the ID of the subscription under your main account, then hit return/enter.
* You should be taken out of the switch process by Subscriptions at this time.
* Confirm a log is created in the latest WCPay log file under WooCommerce > Status > Logs or under `wp-content/uploads/wc_logs`. The log should be similar to: `NOTICE User (49) attempted to switch a subscription (852) not assigned to them.`

-------------------


**Post merge**
